src/tablecloth/api/join_separate.clj

```diff
@@ -62,11 +62,20 @@
        (process-join-columns ds target-column join-function col-names drop-columns?)))))
 
 ;;
-
 (defn- infer-target-columns
   [col res]
-  (let [colname (col/column-name col)]
-    (map #(str colname "-" %) (range (count (first res))))))
+  (let [colname (col/column-name col)
+        update-f (fn [colname value]
```

Comment:
Look! This function is actually the same as `prefix`. So I propose to reuse it.
BTW: internal function which doesn't closure any local bindings is a candidate to extract to a separate `defn`.